src/dbus_api/api.rs

```diff
@@ -193,15 +193,16 @@ fn get_base_tree<'a>(dbus_context: DbusContext) -> (Tree<MTFn<TData>, TData>, db
 fn register_pool_dbus(
     dbus_context: &DbusContext,
     pool_uuid: PoolUuid,
-    pool: &Pool,
+    pool: &mut Pool,
     object_path: &dbus::Path<'static>,
 ) {
-    let pool_path = create_dbus_pool(dbus_context, object_path.clone(), pool_uuid);
-    for (_, fs_uuid, _) in pool.filesystems() {
-        create_dbus_filesystem(dbus_context, pool_path.clone(), fs_uuid);
+    let pool_path = create_dbus_pool(dbus_context, object_path.clone(), pool_uuid, pool);
+    pool.set_dbus_path(object_path.clone());
```

Comment:
This line needs to be removed. ```create_dbus_pool``` has already set the object path, this line resets it to the wrong object path.